tests/endpoint.rs

```diff
@@ -0,0 +1,126 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR BSD-3-Clause
+
+#![cfg(feature = "remote_endpoint")]
+
+mod subscribers;
+
+use std::any::Any;
+use std::result;
+use std::sync::atomic::{AtomicBool, AtomicU64, Ordering};
+use std::sync::Arc;
+use std::thread;
+use std::time::Duration;
+
+use event_manager::{self, EventManager, MutEventSubscriber, SubscriberId};
+use subscribers::CounterSubscriber;
+
+trait SpecialSubscriber: MutEventSubscriber {
+    fn as_mut_any(&mut self) -> &mut dyn Any;
+}
+
+impl SpecialSubscriber for CounterSubscriber {
+    fn as_mut_any(&mut self) -> &mut dyn Any {
+        self
+    }
+}
+
+#[test]
+fn test_endpoint() {
+    let mut event_manager = EventManager::<Box<dyn SpecialSubscriber + Send>>::new().unwrap();
+    let endpoint = event_manager.remote_endpoint();
+    let sub = Box::new(CounterSubscriber::new());
+
+    let run_count = Arc::new(AtomicU64::new(0));
+    let keep_running = Arc::new(AtomicBool::new(true));
+
+    // These get moved into the following closure.
+    let run_count2 = run_count.clone();
```

Comment:
nit: can we use `_clone` instead of 2? It speaks more about what the variable is.